src/Extension.Tests/OutputEvaluationTests.cs

```diff
@@ -118,10 +110,7 @@ public void no_submission_recieved()
         public async Task teacher_can_fail_rule_evaluation_and_provide_feedback_and_hint()
         {
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed